audience_strategy.go

```diff
@@ -92,10 +92,10 @@ func GetAudiences(form url.Values) []string {
 	}
 }
 
-func (f *Fosite) validateAuthorizeAudience(ctx context.Context, r *http.Request, request *AuthorizeRequest) error {
-	audience := GetAudiences(request.Form)
+func (f *Fosite) validateAudience(ctx context.Context, r *http.Request, request Requester) error {
+	audience := GetAudiences(request.GetRequestForm())
 
-	if err := f.Config.GetAudienceStrategy(ctx)(request.Client.GetAudience(), audience); err != nil {
+	if err := f.Config.GetAudienceStrategy(ctx)(request.GetClient().GetAudience(), audience); err != nil {
```

Comment:
didn't check what happens here in GetClient but wonder how it fails  in case it s nil

Reply:
Oh it's exactly the same behavior
```
func (a *Request) GetClient() Client {
	return a.Client
}
```

Reply:
good question.. what does GetClient() return.. if it is an interface.. it can still be nil.. then calling GetAudience() would panic..

Reply:
That's fair. If someone calls this function without setting the client in this request, they will get an panic error (this was the case before these changes). 

The library assumes that you have correctly set the request before calling these functions. I would rather we made the check, but I see no problem with this since they are private functions.